Update Playing Videos page for Godot 4.5

berarma opened the pull request:
Removed another limitation: change playback speed.

~~Added Shutter Encoder as a FFmpeg GUI. Related to #10942.~~

I've decided to remove the date from the FFmpeg warning since they have already released a point version but without the fix. And I don't know how many more they will release yet. I hoped that the next major version was closer to being released.

Changed the GOP description a bit to hopefully better fit the rest of the documentation.

berarma commented:
> What specifically in this PR only applies to 4.5?

I've removed some limitations. I've taken the opportunity to rework some of the text I did previously, and reference Shutter Encoder.

berarma commented:
Oh, sorry, I had forgot about that PR. There's another limitation to remove. I'll do it in this PR.

It's marked as a draft because one of the new features wasn't yet merged. Now that it is I'll update this PR.

The reason I've reworked some of my previous additions to this page is that I feel a bit insecure writing documentation. When I read some parts of it, it feels too verbose or more intricate than necessary. Your judgement on this would help me. Thanks.

berarma commented:
Description updated.

berarma commented:
After testing Shutter Encoder myself I've decided against including it in the docs because [it uses the worst possible encoding parameters and offers no way to change them](https://github.com/paulpacifico/shutter-encoder/issues/298).

skyace65 commented on tutorials/animation/playing_videos.rst:
```diff
@@ -202,10 +200,9 @@ below with almost any input video format (AVI, MOV, WebM, …).
-   due to a couple of bugs. It's highly recommended to use one of the latest
-   static daily builds, or build FFmpeg from their master branch where they're
-   already fixed.
+   Current official FFmpeg releases have some bugs in their Ogg/Theora
+   multiplexer. It's highly recommended to use one of the latest static daily
+   builds, or build from their master branch where they are already fixed.
```
Could you please specify an exact version here? Something like "at the time of writing daily builds are fixed, anything newer than 7.1.1 should be fine." But it doesn't have to be worded like that exactly. Just something so that in the future people aren't wondering what exact version fixed the issue.

berarma replied:
I don't know. It could be fixed in a hypotetical 7.1.2 (less likely) in 7.2.0 (more likely?), or in 8.0 (most probably). I don't know how the release process works. I avoided mentioning versions because of this.

The FFmpeg project page recommends that end users use the git source anyway because it will have the latest fixes. The releases are meant for repackagers and integrators. Maybe we should do the same.

berarma replied:
This is from ffmpeg.org:

> Approximately every 6 months the FFmpeg project makes a new major release. Between major releases point releases will appear that add important bug fixes but no new features. Note that these releases are intended for distributors and system integrators. Users that wish to compile from source themselves are strongly encouraged to consider using the development branch (see above), this is the only version on which FFmpeg developers actively work. The release branches only cherry pick selected changes from the development branch, which therefore receives much more and much faster bug fixes such as additional features and security patches. 

Maybe saying something along the lines of "it's recommended to build from the sources using the development branch or use the daily builds to get the latest features and fixes" as a permanent disclaimer. What do you think?

skyace65 replied:
How about this, do we have a link to the (presumably closed) theora issue in their repository? Or whatever PR fixed it? If we do I'd like it to be linked in a doc comment. That way in 6 months or whenever we can check to see "is this in a release yet?" I just don't like idea of an indefinite "build from source" warning with no easy way to see if the issue fix has been included in a release yet.

skyace65 replied:
I think it's too much to have them as regular text. But as a doc comment so maintainers can see it should be fine. Since FFmpeg doesn't have any specific way for us to tell based on the issue I'd say make a doc comment that links to the commits. In the future we'd be able to download the source code for whatever the newest release is and check if the file changes from the commits are there. It's not easy, but not exactly hard either.

skyace65 reviewed (changes requested):
Looks good! Only one minor change needs to be made.

berarma commented:
I've moved the doc comment below the warning that it refers to so that it's more clear where it belongs. I've added the commit refs to the comment and a minor rewording of the warning.
